common/worker_versioning/worker_versioning.go

```diff
@@ -41,14 +42,20 @@ import (
 )
 
 const (
-	buildIdSearchAttributePrefixAssigned    = "assigned"
-	buildIdSearchAttributePrefixVersioned   = "versioned"
-	buildIdSearchAttributePrefixUnversioned = "unversioned"
-	BuildIdSearchAttributeDelimiter         = ":"
+	buildIdSearchAttributePrefixReachability = "reachability"
+	buildIdSearchAttributePrefixAssigned     = "assigned"
+	buildIdSearchAttributePrefixVersioned    = "versioned"
+	buildIdSearchAttributePrefixUnversioned  = "unversioned"
+	BuildIdSearchAttributeDelimiter          = ":"
 	// UnversionedSearchAttribute is the sentinel value used to mark all unversioned workflows
 	UnversionedSearchAttribute = buildIdSearchAttributePrefixUnversioned
 )
 
+// ReachabilityBuildIdSearchAttribute returns the search attribute value for the currently assigned build ID
+func ReachabilityBuildIdSearchAttribute(behavior enumspb.VersioningBehavior, deployment *deploymentpb.Deployment) string {
+	return fmt.Sprintf("%s:%s:%s", buildIdSearchAttributePrefixReachability, behavior.String(), DeploymentToString(deployment))
+}
```

Comment:
nit - 
I pulled your code locally and noticed the function definition for `DeploymentToString` containing:
`return deployment.SeriesName + ":" + deployment.GetBuildId()`

we can maybe be consistent and just do `deployment.BuildId`

Reply:
lol Shahab wrote that one, but I agree, it should be consistent -- will fix :)